agent/provider/src/cli/rule.rs

```diff
@@ -1,30 +1,64 @@
-use crate::rules::{CertRule, OutboundRule};
-use crate::{
-    rules::{Mode, RulesManager},
-    startup_config::ProviderConfig,
-};
-use anyhow::Result;
+use crate::rules::outbound::{CertRule, Mode, OutboundRules};
+use crate::rules::restrict::{RestrictRule, RuleAccessor};
+use crate::rules::OutboundRule;
+use crate::{rules::RulesManager, startup_config::ProviderConfig};
+
+use anyhow::{bail, Result};
+use serde_json::Value;
 use std::collections::HashMap;
 use std::path::PathBuf;
 use structopt::StructOpt;
 use strum::VariantNames;
+
+use ya_client_model::NodeId;
 use ya_manifest_utils::keystore::{AddParams, AddResponse, Keystore};
 use ya_manifest_utils::short_cert_ids::shorten_cert_ids;
+use ya_manifest_utils::CompositeKeystore;
 use ya_utils_cli::{CommandOutput, ResponseTable};
 
 #[derive(StructOpt, Clone, Debug)]
 pub enum RuleCommand {
     /// Set Modes for specific Rules
     Set(SetRule),
+    /// Add new rule.
+    Add(AddRule),
+    /// Remove existing rule.
+    Remove(RemoveRule),
+    /// Enable all rules in category.
+    Enable(RuleCategory),
+    /// Disable all rules in category.
+    Disable(RuleCategory),
     /// List active Rules and their information
     List,
 }
 
+/// Left for compatibility only. Should be replaced by AddRule and RemoveRule.
 #[derive(StructOpt, Clone, Debug)]
 pub enum SetRule {
     Outbound(SetOutboundRule),
 }
 
+#[derive(StructOpt, Clone, Debug)]
+pub enum RuleCategory {
+    Outbound,
+    Blacklist,
+    AllowOnly,
+}
+
+#[derive(StructOpt, Clone, Debug)]
+pub enum AddRule {
+    Outbound(SetOutboundRule),
+    Blacklist(RestrictRuleDesc),
+    AllowOnly(RestrictRuleDesc),
+}
+
+#[derive(StructOpt, Clone, Debug)]
+pub enum RemoveRule {
+    Outbound(SetOutboundRule),
+    Blacklist(RestrictRuleDesc),
+    AllowOnly(RestrictRuleDesc),
```

Comment:
Maybe there should be some comment/help message indicating a difference between `Outbound` and `Blacklist` `AllowOnly`. Something small indicating that `Outbound` is a more specific rule about nodes in context of Outbound network feature, and `Blacklist` `AllowOnly` are more general rules about allowing/blocking certain nodes.
To be fair I have no idea how to describe it.

Reply:
Documentation will clear this up. Issue: https://github.com/golemfactory/yagna/issues/3137